connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -128,6 +143,26 @@ private[sql] class AvroDeserializer(
       case (INT, IntegerType) => (updater, ordinal, value) =>
         updater.setInt(ordinal, value.asInstanceOf[Int])
 
+      case (INT, DateType) if isNotExactType =>
```

Comment:
The avro connector supports reading int/long as date type for many releases. Why do we need to forbid it?

Reply:
We may get garbage data since int/long is not logically a date type.

Reply:
We want to block this and only allow read DateType as DateType.

Reply:
I am -1 on this. Why do we have to block the conversion?